pkg/apis/pipeline/v1beta1/task_validation.go

```diff
@@ -110,6 +112,46 @@ func validateDeclaredWorkspaces(workspaces []WorkspaceDeclaration, steps []Step,
 	return errs
 }
 
+// validateWorkspaceUsages checks that all WorkspaceUsage objects in Steps
+// refer to workspaces that are defined in the Task.
+//
+// This is an alpha feature and will fail validation if it's used by a step
+// or sidecar when the enable-api-fields feature gate is anything but "alpha".
+func validateWorkspaceUsages(ctx context.Context, ts *TaskSpec) (errs *apis.FieldError) {
+	workspaces := ts.Workspaces
+	steps := ts.Steps
+	sidecars := ts.Sidecars
+
+	wsNames := sets.NewString()
+	for _, w := range workspaces {
+		wsNames.Insert(w.Name)
+	}
+
+	for stepIdx, step := range steps {
+		if len(step.Workspaces) != 0 {
+			errs = errs.Also(ValidateEnabledAPIFields(ctx, "step workspaces", config.AlphaAPIFields).ViaIndex(stepIdx).ViaField("steps"))
+		}
```

Comment:
Should we return an error here or continue validating workspaces even when a feature gate is not set to `alpha`? 🤔 Generally we accumulate all the validation errors before exiting.

Reply:
Yeah, I've decided to leave them accumulating. We can always return early in future if we see reports that it's confusing this way.